core/src/main/java/bisq/core/trade/protocol/bsq_swap/tasks/buyer/ProcessBsqSwapFinalizeTxRequest.java

```diff
@@ -39,20 +51,58 @@ public ProcessBsqSwapFinalizeTxRequest(TaskRunner<BsqSwapTrade> taskHandler, Bsq
     @Override
     protected void run() {
         try {
-            runInterceptHook();
-
             BsqSwapFinalizeTxRequest request = checkNotNull((BsqSwapFinalizeTxRequest) protocolModel.getTradeMessage());
             checkNotNull(request);
             Validator.checkTradeId(protocolModel.getOfferId(), request);
 
-            BsqSwapTradePeer tradePeer = protocolModel.getTradePeer();
+            // We will use only the sellers inputs from the tx so we do not verify anything else
+            byte[] tx = request.getTx();
+            Transaction sellersTransaction = protocolModel.getBtcWalletService().getTxFromSerializedTx(tx);
+            List<RawTransactionInput> rawInputs = request.getBtcInputs();
```

Comment:
Would be good to verify that these are correct BTC inputs if a local bitcoin node is available.

Reply:
I think that adds too much complexity. Worst case is the tx is invalid and the user need to do a spv resync.